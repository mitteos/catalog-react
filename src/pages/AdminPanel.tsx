import React from 'react';
import styled from "styled-components";
import {ProductForm, ProductList} from "components/AdminPanel";

export const AdminPanel = () => {
    return (
        <Container>
            <Title>Админ панель</Title>
            <Subtitle>Добавление товара</Subtitle>
            <ProductForm isNewProduct={true}/>
            <Subtitle>Редактирование товаров</Subtitle>
            <ProductList />
        </Container>
    );
};

const Container =  styled.div`
  max-width: 1369px;
  margin: 0 auto;
  width: 71.3%;
  display: flex;
  flex-direction: column;
  gap: 50px;
  @media (max-width: 1200px) {
    width: 90%;
  }
`
const Title = styled.h1`
  font-weight: 500;
  font-size: 40px;
  line-height: 120%;
  text-transform: uppercase;
  color: #111111;
  @media (max-width: 700px) {
    font-size: 30px;
  }
  @media (max-width: 500px) {
    font-size: 20px;
  }
`
const Subtitle = styled.p`
  text-transform: capitalize;
  font-size: 25px;
  margin: 0 0 10px;
  @media (max-width: 700px) {
    font-size: 20px;
  }
  @media (max-width: 500px) {
    font-size: 18px;
  }
`
