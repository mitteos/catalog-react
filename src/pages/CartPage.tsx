import React from 'react';
import styled from "styled-components";
import {Crumbs} from "components/Layout";
import {CartList} from "components/Products";
import {AccentButton} from "components/UI";
import {Link} from "react-router-dom";
import {useAppSelector} from "hooks/redux";

export const CartPage = () => {

    const {items} = useAppSelector(state => state.cart)
    const totalPrice = items?.map(el => el.product.price * el.count).reduce((prev, next) => prev + next, 0)

    return (
        <Container data-testid="cart-page">
            <Crumbs>
                <CrumbItem>Корзина</CrumbItem>
            </Crumbs>
            <Title>Корзина</Title>
            <Separator />
            {items?.length
                ? <CartList />
                : <Empty data-testid="empty-list">Корзина пуста</Empty>
            }
            {!!items?.length &&
                <Total>
                    <Link to="/cart/checkout">
                        <Checkout>
                            <p>Оформить заказ</p>
                        </Checkout>
                    </Link>
                    <TotalPrice>{totalPrice} ₸</TotalPrice>
                </Total>
            }
        </Container>
    );
};
const Container =  styled.div`
  max-width: 1369px;
  margin: 0 auto;
  width: 71.3%;
  @media (max-width: 1200px) {
    width: 90%;
  }
`
const CrumbItem = styled.p`
  font-weight: 300;
  font-size: 14px;
  line-height: 120%;
  color: #3F4E65;
  opacity: 0.5;
`
const Title = styled.h1`
  font-weight: 500;
  font-size: 40px;
  line-height: 120%;
  text-transform: uppercase;
  color: #111111;
  @media (max-width: 700px) {
    font-size: 30px;
  }
  @media (max-width: 500px) {
    font-size: 20px;
  }
`
const Separator = styled.div`
  width: 100%;
  opacity: 0.3;
  border: 1px dashed #3F4E65;
  margin: 50px 0;
  @media (max-width: 500px) {
    margin: 20px 0;
  }
`
const Empty = styled.p`
  font-weight: 500;
  font-size: 20px;
  text-align: center;
  line-height: 130%;
  color: #111111;
`
const Total = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 50px 0 0 0;
  @media (max-width: 500px) {
    flex-direction: column;
    gap: 25px;
    margin: 25px 0 0 0;
  }
`
const Checkout = styled(AccentButton)`
  padding: 21px 37px;
  p {
    font-weight: 700;
    font-size: 14px;
    line-height: 17px;
    color: #FFFFFF;
  }
  @media (max-width: 500px) {
    width: 100%;
    order: 2;
  }
`
const TotalPrice = styled.p`
  font-weight: 700;
  font-size: 30px;
  line-height: 130%;
  color: #111111;
  @media (max-width: 500px) {
    order: 1;
  }
`
