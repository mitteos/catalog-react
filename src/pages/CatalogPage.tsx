import React from 'react';
import styled from "styled-components";
import {CategorySort, FilterBar, SelectSort} from "components/Catalog";
import {ProductList} from "components/Products";
import {Crumbs} from "components/Layout";

export const CatalogPage = () => {
    return (
        <Container data-testid="main-page">
            <Crumbs>
                <CrumbItem>Косметика и гигиена</CrumbItem>
            </Crumbs>
            <Grid>
                <HeaderTitle>Косметика и гигиена</HeaderTitle>
                <HeaderSelect />
                <Categories />
                <Filter />
                <Products />
            </Grid>
        </Container>
    );
};

const Container =  styled.div`
  max-width: 1369px;
  margin: 0 auto;
  width: 71.3%;
  @media (max-width: 1200px) {
    width: 90%;
  }
`
const CrumbItem = styled.p`
  font-weight: 300;
  font-size: 14px;
  line-height: 120%;
  color: #3F4E65;
  opacity: 0.5;
`
const HeaderTitle = styled.p`
  font-weight: 500;
  font-size: 40px;
  line-height: 120%;
  text-transform: uppercase;
  color: #111111;
  grid-column: 1/3;
  margin: 0 0 -5px;
  @media (max-width: 900px) {
    grid-column: 1/4;
    grid-row: 1/2;
  }
  @media (max-width: 700px) {
    font-size: 30px;
  }
  @media (max-width: 500px) {
    font-size: 20px;
  }
`
const HeaderSelect = styled(SelectSort)`
  grid-column: 3/4;
  justify-content: flex-end;
  margin: 0 0 -5px;
  @media (max-width: 900px) {
    grid-column: 1/4;
    grid-row: 4/5;
    justify-content: flex-start;
  }
`
const Grid = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: 238px 1fr auto;
  grid-template-rows: auto auto;
  gap: 30px;
  @media (max-width: 900px) {
    grid-template-columns: auto;
    grid-template-rows: repeat(5, auto);
  }
`
const Categories = styled(CategorySort)`
  grid-column: 1/4;
  @media (max-width: 900px) {
    grid-column: 1/4;
    grid-row: 3/4;
    flex-direction: column;
    gap: 10px;
  }
`
const Filter = styled(FilterBar)`
  @media (max-width: 900px) {
    grid-column: 1/4;
    grid-row: 2/3;
    max-width: none;
  }`
const Products = styled(ProductList)`
  grid-column: 2/4;
  @media (max-width: 900px) {
  grid-column: 1/4;
  grid-row: 5/6;
}
`
