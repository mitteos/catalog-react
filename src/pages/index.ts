export {CartPage} from "./CartPage"
export {CatalogPage} from "./CatalogPage"
export {CheckoutPage} from "./CheckoutPage"
export {ProductPage} from "./ProductPage"
export {AdminPanel} from "./AdminPanel"
export {AdminEditProduct} from "./AdminEditProduct"
