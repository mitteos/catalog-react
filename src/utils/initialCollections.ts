import {ProductState} from "store/features/product/types";
import Product4 from "assets/img/product4.png";
import Product3 from "assets/img/product3.png";
import Product2 from "assets/img/product2.png";
import Product1 from "assets/img/product1.png";
import {BrandState} from "store/features/brand/types";
import {TypeState} from "store/features/type/types";

export const initialProductsCollections: ProductState[] = [
    {
        id: 1,
        name: "Жидкость для мытья посуды",
        brandId: 1,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 300,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "Max",
        vendor: 521,
        img: Product4
    },
    {
        id: 2,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product2
    },
    {
        id: 3,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product3
    },
    {
        id: 4,
        name: "Жидкость для мытья посуды",
        brandId: 2,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 350,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "AOS",
        vendor: 523,
        img: Product1
    },
    {
        id: 5,
        name: "Жидкость для мытья посуды",
        brandId: 1,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 300,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "Max",
        vendor: 521,
        img: Product4
    },
    {
        id: 6,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product2
    },
    {
        id: 7,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product3
    },
    {
        id: 8,
        name: "Жидкость для мытья посуды",
        brandId: 2,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 350,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "AOS",
        vendor: 523,
        img: Product1
    },
    {
        id: 9,
        name: "Жидкость для мытья посуды",
        brandId: 1,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 300,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "Max",
        vendor: 521,
        img: Product4
    },
    {
        id: 10,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product2
    },
    {
        id: 11,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product3
    },
    {
        id: 12,
        name: "Жидкость для мытья посуды",
        brandId: 2,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 350,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "AOS",
        vendor: 523,
        img: Product1
    },
    {
        id: 13,
        name: "Жидкость для мытья посуды",
        brandId: 1,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 300,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "Max",
        vendor: 521,
        img: Product4
    },
    {
        id: 14,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product2
    },
    {
        id: 15,
        name: "Порошок для стирки",
        brandId: 4,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [2],
        price: 300,
        sizeType: "weight",
        sizeValue: 500,
        manufacturer: "Ariel",
        vendor: 521,
        img: Product3
    },
    {
        id: 16,
        name: "Жидкость для мытья посуды",
        brandId: 2,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 350,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "AOS",
        vendor: 523,
        img: Product1
    },
    {
        id: 17,
        name: "Жидкость для мытья посуды",
        brandId: 2,
        barcode: "32623621",
        description: "Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл",
        typeId: [1],
        price: 350,
        sizeType: "volume",
        sizeValue: 500,
        manufacturer: "AOS",
        vendor: 523,
        img: Product1
    },
]

export const initialBrandsCollection: BrandState[] = [
    {id: 1, name: "AOS"},
    {id: 2, name: "MAX"},
    {id: 3, name: "ARIEL"},
    {id: 4, name: "BIMAX"},
]

export const initialTypesCollection: TypeState[] = [
    {id: 1, name: "Уход за кожей"},
    {id: 2, name: "Уход за руками"},
    {id: 3, name: "Гигиена полости рта"},
    {id: 4, name: "Подарочные наборы"},
]
