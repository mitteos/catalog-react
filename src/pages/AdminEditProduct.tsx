import React from 'react';
import styled from "styled-components";
import {useParams} from "react-router-dom";
import {ProductForm} from "components/AdminPanel";
import {useAppSelector} from "hooks/redux";

export const AdminEditProduct = () => {

    const {id} = useParams()
    const {products} = useAppSelector(state => state.product)

    return (
        <Container>
            <Title>Редактирование товара</Title>
            <ProductForm
                productInfo={products.find(el => id && el.id === +id)}
                isNewProduct={false}/>
        </Container>
    );
};

const Container =  styled.div`
  max-width: 1369px;
  margin: 0 auto;
  width: 71.3%;
  display: flex;
  flex-direction: column;
  gap: 50px;
  @media (max-width: 1200px) {
    width: 90%;
  }
`
const Title = styled.h1`
  font-weight: 500;
  font-size: 40px;
  line-height: 120%;
  text-transform: uppercase;
  color: #111111;
  @media (max-width: 700px) {
    font-size: 30px;
  }
  @media (max-width: 500px) {
    font-size: 20px;
  }
`
