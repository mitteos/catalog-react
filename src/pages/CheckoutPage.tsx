import React, {useState} from 'react';
import styled from "styled-components";
import {Crumbs} from "components/Layout";
import {AccentButton, DropdownInput, TextArea, TextInput} from "components/UI";
import {InfoList, ProductList, SuccessModal} from "components/Checkout";
import {useAppDispatch} from "hooks/redux";
import {cartActions} from "store/features/cart";
import {cities} from "utils/cities";
import {SubmitHandler, useForm} from "react-hook-form";
import {Link} from "react-router-dom";

interface CheckoutInputs {
    name: string
    phone: string
    email: string
    organization: string
    address: string
    comment: string
}

export const CheckoutPage = () => {

    const [isModalActive, setIsModalActive] = useState<boolean>(false)
    const dispatch = useAppDispatch()
    const [addressValue, setAddressValue] = useState({id: 0, name: "Выберите ваш город"} as {id: number, name: string})
    const {register, handleSubmit, formState: { errors }} = useForm<CheckoutInputs>()


    const makeOrder: SubmitHandler<CheckoutInputs> = (formFields) => {
        console.log(formFields);
        setIsModalActive(true)
        dispatch(cartActions.clearItems())
    }

    return (
        <>
            <Container>
                <Crumbs isCheckout={true}>
                    <CrumbsItem to="/cart">Корзина</CrumbsItem>
                    <CrumbSeparator />
                    <CrumbItemLast>Оформление заказа</CrumbItemLast>
                </Crumbs>
                <Title>Оформление заказа</Title>
                <Row onSubmit={handleSubmit(makeOrder)}>
                    <Column>
                        <InfoPart>
                            <InfoPartHeader>
                                <InfoPartNumber>1</InfoPartNumber>
                                <InfoPartTitle>Контактные данные</InfoPartTitle>
                            </InfoPartHeader>
                            <TextInput
                                title="Имя*"
                                placeholder="Введите ваше имя"
                                name="name"
                                register={register}
                                errors={errors.name}
                                required={true}
                            />
                            <TextInput
                                title="Телефон*"
                                placeholder="Введите ваш телефон"
                                name="phone"
                                register={register}
                                errors={errors.phone}
                                required={true}
                            />
                            <TextInput
                                title="E-mail*"
                                placeholder="Введите ваш E-mail"
                                name="email"
                                register={register}
                                errors={errors.email}
                                required={true}
                            />
                            <TextInput
                                title="Название организации"
                                placeholder="Введите название организации"
                                name="organization"
                                register={register}
                                errors={errors.organization}
                            />
                        </InfoPart>
                        <InfoPart>
                            <InfoPartHeader>
                                <InfoPartNumber>2</InfoPartNumber>
                                <InfoPartTitle>адрес доставки</InfoPartTitle>
                            </InfoPartHeader>
                            <DropdownInput title="Город" headValue={addressValue.name}>
                                {cities.map(city =>
                                    <DropDownItem
                                        key={city.id}
                                        onClick={() => setAddressValue(city)}
                                    >{city.name}</DropDownItem>
                                )}
                            </DropdownInput>
                            <TextInput
                                title="Адрес"
                                placeholder="Введите адрес доставки"
                                name="address"
                                register={register}
                                errors={errors.address}
                            />
                        </InfoPart>
                        <ConfirmBrn onClick={handleSubmit(makeOrder)}>Подтверждение заказа</ConfirmBrn>
                    </Column>
                    <ColumnAddition>
                        <InfoPart>
                            <InfoList />
                        </InfoPart>
                        <InfoPart>
                            <InfoPartHeader>
                                <InfoPartNumber>3</InfoPartNumber>
                                <InfoPartTitle>Дополнительно</InfoPartTitle>
                            </InfoPartHeader>
                            <TextArea
                                title="Комментарий"
                                placeholder="Введите ваш комментарий"
                                name="comment"
                                register={register}
                                errors={errors.comment}
                            />
                        </InfoPart>
                    </ColumnAddition>
                    <ColumnOrder>
                        <ProductList />
                        <CheckoutBtn onClick={handleSubmit(makeOrder)}>Подтверждение заказа</CheckoutBtn>
                    </ColumnOrder>
                </Row>
            </Container>
            <SuccessModal isActive={isModalActive} setIsActive={setIsModalActive}/>
        </>
    );
};

const Container =  styled.div`
  max-width: 1369px;
  margin: 0 auto;
  width: 71.3%;
  @media (max-width: 1200px) {
    width: 90%;
  }
`
const CrumbsItem = styled(Link)`
  font-weight: 300;
  font-size: 14px;
  line-height: 120%;
  color: #3F4E65;
`
const CrumbItemLast = styled.p`
  font-weight: 300;
  font-size: 14px;
  line-height: 120%;
  color: #3F4E65;
  opacity: 0.5;
`
const CrumbSeparator = styled.div`
  height: 16px;
  opacity: 0.3;
  border: 1px dashed #3F4E65;
`
const Title = styled.h1`
  font-weight: 500;
  font-size: 40px;
  line-height: 120%;
  text-transform: uppercase;
  color: #111111;
  @media (max-width: 700px) {
    font-size: 30px;
  }
  @media (max-width: 500px) {
    font-size: 20px;
  }
`
const Row = styled.form`
  display: flex;
  justify-content: space-between;
  margin: 50px 0 0 0;
  gap: 20px;
  @media (max-width: 1500px) {
    flex-wrap: wrap;
    justify-content: center;
  }
  @media (max-width: 664px) {
    width: 100%;
    margin: 26px 0 0 0;
  }
`
const Column = styled.div`
  display: flex;
  flex-direction: column;
  gap: 40px;
  @media (max-width: 664px) {
    width: 100%;
    gap: 30px;
  }
`
const ColumnOrder = styled(Column)`
  gap: 0;
  width: 569px;
  @media (max-width: 1650px) {
    flex: auto;
  }
`
const CheckoutBtn = styled(AccentButton)`
  display: none;
  @media (max-width: 664px) {
    display: flex;
    font-weight: 700;
    font-size: 14px;
    line-height: 17px;
    color: #FFFFFF;
    padding: 31px 0;
    margin: 30px 0 0 0;
  }
`
const ColumnAddition = styled(Column)`
  @media (max-width: 664px) {
    flex-direction: column-reverse;
  }
`
const InfoPart = styled.div`
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 289px;
  @media (max-width: 664px) {
    width: 100%;
  }
`
const InfoPartHeader = styled.div`
  display: flex;
  align-items: center;
  gap: 15px;
  margin: 0 0 15px;
`
const DropDownItem = styled.p`
  font-weight: 500;
  font-size: 14px;
  line-height: 150%;
  color: #111111;
`
const InfoPartTitle = styled.p`
  font-weight: 500;
  font-size: 16px;
  line-height: 150%;
  text-transform: uppercase;
  color: #111111;
`
const InfoPartNumber =styled.p`
  width: 35px;
  padding: 8.5px 0;
  text-align: center;
  border-radius: 100%;
  font-weight: 500;
  font-size: 14px;
  line-height: 130%;
  color: #3F4E65;
  background: linear-gradient(90deg, rgba(255, 198, 80, 0.3) 0%, rgba(254, 202, 110, 0.3) 97.25%);
`
const ConfirmBrn = styled(AccentButton)`
  width: 100%;
  padding: 21px 0;
  font-weight: 700;
  font-size: 14px;
  line-height: 17px;
  color: #FFFFFF;
  @media (max-width: 664px) {
    display: none;
  }
`
