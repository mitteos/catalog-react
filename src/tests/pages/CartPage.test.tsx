import * as ReduxHooks from "react-redux";
import {initialProductsCollections} from "utils/initialCollections";
import {CartPage} from "pages";
import {screen} from "@testing-library/react";
import {renderWithRouter} from "tests/helpers/renderWithRouter";

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

const mockState = {
    items: [
        {product: initialProductsCollections[0], count: 2},
        {product: initialProductsCollections[3], count: 6},
    ]
}

describe("cart page", () => {
    test("render with products", () => {
        jest.spyOn(ReduxHooks, "useSelector").mockReturnValue(mockState)
        renderWithRouter(<CartPage />)
        expect(screen.getByTestId("cart-page")).toMatchSnapshot()
    })
    test("render empty list", () => {
        jest.spyOn(ReduxHooks, "useSelector").mockReturnValue({items: []})
        renderWithRouter(<CartPage />)
        expect(screen.getByTestId("empty-list")).not.toBeNull()
    })
})
