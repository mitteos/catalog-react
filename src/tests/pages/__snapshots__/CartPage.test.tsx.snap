// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart page render with products 1`] = `
<div
  class="sc-iXRnGQ hQrAHU"
  data-testid="cart-page"
>
  <div>
    <div
      class="sc-gKcDQK fhhudC"
    >
      <div
        class="sc-fzmOIZ bJyZzB"
      >
        <img
          alt=""
          src="arrowBtn.svg"
        />
      </div>
      <p>
        назад
      </p>
    </div>
    <div
      class="sc-knMnlt ekYAgR"
    >
      <a
        class="sc-cwVcKo cFcsNv"
        href="/"
      >
        Главная
      </a>
      <div
        class="sc-dlyeST ja-Dpgk"
      />
      <p
        class="sc-jMupPv NqbYW"
      >
        Корзина
      </p>
    </div>
  </div>
  <h1
    class="sc-bVDfoq cwQvzB"
  >
    Корзина
  </h1>
  <div
    class="sc-rZqKh cOhYHe"
  />
  <div
    class="sc-jJgynQ jcMAxX"
  >
    <div
      class="sc-jJgynQ jcMAxX"
    >
      <div
        class="sc-oQLfA hQIZVY"
      >
        <div
          class="sc-hjsqBZ iobvah"
        >
          <img
            alt=""
            src="product4.png"
          />
        </div>
        <div
          class="sc-ejdXBC caa-DCy"
        >
          <div
            class="sc-gHyekI jYFqpQ"
          >
            <img
              alt=""
              src="volume.svg"
            />
            <p>
              500
               
              мл
            </p>
          </div>
          <p
            class="sc-cmEail cJPSx"
          >
             
            Жидкость для мытья посуды
          </p>
          <p
            class="sc-eZYOHW cuUQZM"
          >
            Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл
          </p>
        </div>
        <div
          class="sc-FeKFz sc-fYdXmn lfOfhh dxZLMj"
        />
        <div
          class="sc-ldFCYb kLRasJ"
        >
          <div
            class="sc-irTswW dQYMys"
          >
            <div
              class="sc-csCMJt dUfkSl"
            >
              -
            </div>
            <p
              class="sc-fbJfA kmQOFp"
            >
              2
            </p>
            <div
              class="sc-csCMJt dUfkSl"
            >
              +
            </div>
          </div>
          <div
            class="sc-FeKFz lfOfhh"
          />
          <p
            class="sc-dxroEu cXBTeX"
          >
            300
             ₸
          </p>
          <div
            class="sc-FeKFz lfOfhh"
          />
          <div
            class="sc-beqWaB hTSdDr sc-UxxwN diIooH"
          >
            <img
              alt=""
              src="basket.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-bbpGSz kvUrVK"
      />
    </div>
    <div
      class="sc-jJgynQ jcMAxX"
    >
      <div
        class="sc-oQLfA hQIZVY"
      >
        <div
          class="sc-hjsqBZ iobvah"
        >
          <img
            alt=""
            src="product1.png"
          />
        </div>
        <div
          class="sc-ejdXBC caa-DCy"
        >
          <div
            class="sc-gHyekI jYFqpQ"
          >
            <img
              alt=""
              src="volume.svg"
            />
            <p>
              500
               
              мл
            </p>
          </div>
          <p
            class="sc-cmEail cJPSx"
          >
             
            Жидкость для мытья посуды
          </p>
          <p
            class="sc-eZYOHW cuUQZM"
          >
            Средства бытовой химии марки Dr MAX NEW: жидкость для мытья посуды - концентрат Таежная земляника 500 мл
          </p>
        </div>
        <div
          class="sc-FeKFz sc-fYdXmn lfOfhh dxZLMj"
        />
        <div
          class="sc-ldFCYb kLRasJ"
        >
          <div
            class="sc-irTswW dQYMys"
          >
            <div
              class="sc-csCMJt dUfkSl"
            >
              -
            </div>
            <p
              class="sc-fbJfA kmQOFp"
            >
              6
            </p>
            <div
              class="sc-csCMJt dUfkSl"
            >
              +
            </div>
          </div>
          <div
            class="sc-FeKFz lfOfhh"
          />
          <p
            class="sc-dxroEu cXBTeX"
          >
            350
             ₸
          </p>
          <div
            class="sc-FeKFz lfOfhh"
          />
          <div
            class="sc-beqWaB hTSdDr sc-UxxwN diIooH"
          >
            <img
              alt=""
              src="basket.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-bbpGSz kvUrVK"
      />
    </div>
  </div>
  <div
    class="sc-ialjHA gqpYIa"
  >
    <a
      href="/cart/checkout"
    >
      <div
        class="sc-beqWaB hTSdDr sc-cIuQVt jQWEhq"
      >
        <p>
          Оформить заказ
        </p>
      </div>
    </a>
    <p
      class="sc-KLwla foHZqZ"
    >
      2700
       ₸
    </p>
  </div>
</div>
`;
